stdlib/sys.pyi

```diff
@@ -27,12 +28,11 @@ _T = TypeVar("_T")
 # The following type alias are stub-only and do not exist during runtime
 _ExcInfo = Tuple[Type[BaseException], BaseException, TracebackType]
 _OptExcInfo = Union[_ExcInfo, Tuple[None, None, None]]
-_PathSequence = Sequence[Union[bytes, str]]
+_ModulePath = MutableSequence[str]  # Same as `types.ModuleType.__path__`
```

Comment:
I'm confused about both of the things changed here :)

- Why are `bytes` paths banned?
- Why does `find_spec()` need to mutate the sequence?

Reply:
> Why are `bytes` paths banned?

Per the documentation the "path will be the value of __path__ from the parent package" which is annotated as being of type `MutableSequence[str]` in the typeshed. https://docs.python.org/3/reference/import.html#path__ also says:

> If __path__ is not empty, it must produce strings when iterated over. More details on the semantics of __path__ are given below.

Essentialy this is a list of `sys.path` entries which can't be `bytes`.

> Why does `find_spec()` need to mutate the sequence?

I suppose it doesn't; I simply copied the annotation from `ModuleType`.  (On that note, why is _that_ mutable?)

Reply:
It can be useful to do `some_package.__path__.append(a_new_path_to_import_from)`, although this is usually done in `__init__.py`, where you can do `__path__.append(blah)` with `__path__` as a global variable. I still don't see why this has to be mutable, but it's fine to use `MutableSequence` for consistency. It would be good to add a comment to `types.ModuleType.__path__` too, so that it stays in sync.